tests/src/Kernel/EventHandlerTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\EventHandlerTest.
+ */
+
+namespace Drupal\Tests\rules\Kernel;
+
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\rules\Context\ContextConfig;
+use Drupal\rules\Event\EntityEvent;
+
+/**
+ * Tests events with qualified name.
+ *
+ * @group rules
+ */
+class EventHandlerTest extends RulesDrupalTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['rules', 'system', 'node', 'field', 'user'];
+
+  /**
+   * The entity storage for Rules config entities.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected $storage;
+
+  /**
+   * A node used for testing.
+   *
+   * @var \Drupal\node\NodeInterface
+   */
+  protected $node;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->installSchema('system', ['sequences']);
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installConfig(['field']);
+
+    $this->storage = $this->container->get('entity_type.manager')->getStorage('rules_reaction_rule');
+
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $entity_type_manager->getStorage('node_type')
+      ->create(['type' => 'page'])
+      ->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'field_integer',
+      'type' => 'integer',
+      'entity_type' => 'node',
+      'cardinality' => FieldStorageDefinitionInterface::CARDINALITY_UNLIMITED,
+    ])->save();
+    FieldConfig::create([
+      'field_name' => 'field_integer',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+    ])->save();
+
+    $this->node = $entity_type_manager->getStorage('node')
+      ->create([
+        'title' => 'test',
+        'type' => 'page',
+      ]);
+  }
+
+  /**
+   * Tests ConfigurableEventHandlerEntityBundle.
+   *
+   * @todo Add integrity check that node.field_integer is detected by Rules.
+   */
+  public function testEntityBundleHandler() {
+    // Create rule1 with the 'rules_entity_presave:node--page' event.
```

Comment:
would should add some high level summary of we are doing: We test that rules are triggered correctly based upon the fully qualified event name as well as the base event name.